core/tee/tadb.c

```diff
@@ -367,6 +323,66 @@ static bool __maybe_unused tadb_check_state(const struct tee_tadb_ta *ta,
 	return false;
 }
 
+static TEE_Result populate_files(struct tee_tadb_dir *db)
+{
+	TEE_Result res;
+	size_t idx;
+	const TEE_UUID null_uuid = { 0 };
+
+	/*
+	 * If db->files isn't NULL the bitfield is already populated and
+	 * there's nothing left to do here for now.
+	 */
+	if (db->files)
+		return TEE_SUCCESS;
+
+	/*
+	 * Iterate over the TA database and set the bits in the bit field
+	 * for used file numbers. Note that set_file() will allocate and
+	 * grow the bitfield as needed.
+	 *
+	 * At the same time clean out duplicate file numbers, the first
+	 * entry with the file number has precedence.  Duplicate entries is
+	 * not  supposed to be able to happen, but if it still does better
+	 * to clean it out here instead of letting the error spread with
+	 * unexpected side effects.
+	 */
+	for (idx = 0;; idx++) {
+		struct tadb_entry entry;
+
+		res = read_ent(db, idx, &entry);
+		if (res) {
+			if (res == TEE_ERROR_ITEM_NOT_FOUND)
```

Comment:
It is confusing, that only one successful (apart from L336) return from this function resides in the middle of it. 
Now, when I grasped idea of this function, this seems obvious. But it was hard to understand from a first try.

This is just a note. I don't ask you to perform any actions.